test/common/upstream/load_balancer_fuzz_base.cc

```diff
@@ -0,0 +1,118 @@
+#include "test/common/upstream/load_balancer_fuzz_base.h"
+
+#include "test/common/upstream/utility.h"
+
+namespace Envoy {
+
+namespace Random {
+uint64_t FakeRandomGenerator::random() {
+  uint8_t index_of_data = counter % bytestring_.size();
+  ++counter;
+  ENVOY_LOG_MISC(trace, "random() returned: {}", bytestring_.at(index_of_data));
+  return bytestring_.at(index_of_data);
+}
+} // namespace Random
+
+namespace Upstream {
+
+// Anonymous namespace for helper functions
+namespace {
+std::vector<uint64_t>
+constructByteVectorForRandom(const Protobuf::RepeatedField<uint64_t>& byte_string) {
+  std::vector<uint64_t> byteVector;
+  byteVector.reserve(byte_string.size());
+  for (const auto eightBytes : byte_string) {
+    byteVector.push_back(eightBytes);
+  }
+  return byteVector;
+}
+} // namespace
+
+void LoadBalancerFuzzBase::initializeFixedHostSets(uint32_t num_hosts_in_priority_set,
+                                                   uint32_t num_hosts_in_failover_set) {
+  int port = 80;
+  for (uint32_t i = 0; i < num_hosts_in_priority_set; ++i) {
+    host_set_.hosts_.push_back(makeTestHost(info_, "tcp://127.0.0.1:" + std::to_string(port)));
+    ++port;
+  }
+  for (uint32_t i = 0; i < num_hosts_in_failover_set; ++i) {
+    failover_host_set_.hosts_.push_back(
+        makeTestHost(info_, "tcp://127.0.0.1:" + std::to_string(port)));
+    ++port;
+  }
+  // TODO(zasweq): More than two host sets?
```

Comment:
Yeah, see my comment above. I think it's fine to start with what you have, but we might want to track generalizing this.